### PR TITLE
test: add listAllTransactions test cases

### PR DESCRIPTION
**Pull request by Andrei199991**
related to #259

### CODE REVIEW

File: packages/wallet/backend/tests/transaction/service.test.ts [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+        createRafikiPaymentPointer: () => ({
+          id: faker.string.uuid(),
+          url: faker.string.alpha(10)
+        })
+      }
+    }
+    Reflect.set(accountService, 'deps', accountServiceDepsMocked)
+    Reflect.set(paymentPointerService, 'deps', ppServiceDepsMocked)
+
+    const account = await accountService.createAccount({
```
Comment (dragosp1011) [inline packages/wallet/backend/tests/transaction/service.test.ts:96]: I suggest to insert the fake account directly in db
in transactions test there is no value in mocking all the things that account service needs to create an account

File: packages/wallet/backend/tests/transaction/service.test.ts [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+  beforeEach(async (): Promise<void> => {
+    res = createResponse()
+    req = createRequest()
+
+    req.body = args
+
+    await createUser({ ...args, isEmailVerified: true })
+    await applyMiddleware(withSession, req, res)
+
+    const { user, session } = await authService.authorize(args)
```
Comment (dragosp1011) [inline packages/wallet/backend/tests/transaction/service.test.ts:133]: extract the code below in a `loginUser` helper method

File: packages/wallet/backend/tests/transaction/service.test.ts [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+    appContainer = await createApp(bindings)
+    knex = appContainer.knex
+    authService = await bindings.resolve('authService')
+    accountService = await bindings.resolve('accountService')
+    paymentPointerService = await bindings.resolve('paymentPointerService')
+    transactionService = await bindings.resolve('transactionService')
+  })
+
+  beforeEach(async (): Promise<void> => {
+    res = createResponse()
```
Comment (dragosp1011) [inline packages/wallet/backend/tests/transaction/service.test.ts:125]: as the service is tested, there is no use for req, res as the user id and other details are passed as params

File: packages/wallet/backend/tests/transaction/service.test.ts [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+  beforeAll(async (): Promise<void> => {
+    bindings = createContainer(env)
+    appContainer = await createApp(bindings)
+    knex = appContainer.knex
+    authService = await bindings.resolve('authService')
+    transactionService = await bindings.resolve('transactionService')
+  })
+
+  beforeEach(async (): Promise<void> => {
+    const res = createResponse()
```
Comment (dragosp1011) [inline packages/wallet/backend/tests/transaction/service.test.ts:59]: request and response are not used in tests. It is enough to insert in db the fake user

File: packages/wallet/backend/tests/transaction/service.test.ts [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+      expect(transactions.total).toEqual(0)
+      expect(transactions.results.length).toEqual(0)
+    })
+
+    it('should list all transactions (4 transactions)', async (): Promise<void> => {
+      const { paymentPointer, account } = await prepareTransactionDependencies()
+
+      await Promise.all(
+        mockedTransactionInsertObjs.map(async (mockedTransactionInsertObj) =>
+          Transaction.query().insert({
```
Comment (dragosp1011) [inline packages/wallet/backend/tests/transaction/service.test.ts:100]: insert is working with arrays as well, so the transactions list can be passed in one go

**Review by dragosp1011 — APPROVED** [review]

**Review by raducristianpopa — APPROVED** [review]

**Pull request merged**